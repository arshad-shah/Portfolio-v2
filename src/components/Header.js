import React, { useState } from "react";
import { AppBar, Toolbar, Typography } from "@material-ui/core";
import HideOnScroll from "./HideOnScroll";

import Desktop from "./navigation/Desktop";
import Mobile from "./navigation/Mobile";
import useStyles from "../styles/Header.styles";
import logo from "../assets/logo.svg";
import PropTypes from "prop-types";

export default function Header(props) {
  const classes = useStyles();
  const { isMobile } = props;

  const [click, setClick] = useState(false);
  const handleClick = () => setClick(!click);
  const closeMobileMenu = () => {
    setClick(false);
  };
  const handleMenu = () => {
    closeMobileMenu();
    handleClick();
  };

  return (
    <div className={classes.root}>
      <Toolbar id="back-to-top-anchor" />
      <HideOnScroll {...props}>
        <AppBar data-testid="header">
          <Toolbar>
            <img aria-label="Logo" src={logo} className={classes.logo} alt="" />
            <Typography
              variant="h5"
              component="p"
              color="textPrimary"
              className={classes.title}
            >
              A SHAH
            </Typography>
            {isMobile ? (
              <Mobile
                click={click}
                closeMobileMenu={closeMobileMenu}
                handleMenu={handleMenu}
              />
            ) : (
              <Desktop />
            )}
          </Toolbar>
        </AppBar>
      </HideOnScroll>
    </div>
  );
}
Header.propTypes = {
  isMobile: PropTypes.bool,
};
