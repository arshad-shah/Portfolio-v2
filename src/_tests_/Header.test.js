import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../components/Header";

describe("Header of the site:", () => {
  it("should render Header with the creater name in it", () => {
    render(<Header />);
    const creator = screen.getByText(/A Shah/i);
    expect(creator).toBeInTheDocument();
  });

  it("should render the logo", () => {
    render(<Header />);
    const Logo = screen.getByLabelText("Logo");
    expect(Logo).toBeInTheDocument();
  });

  it("should render the Mobile menu", () => {
    render(<Header isMobile />);
    const menu = screen.getByLabelText("mobileMenu");
    expect(menu).toBeInTheDocument();
  });

  it("should open mobile menu when menu button is clicked", () => {
    render(<Header isMobile />);
    const menu = screen.getByLabelText("mobileMenu");
    menu.click();
    expect(screen.getByText(/About me/i)).toBeInTheDocument();
    expect(screen.getByText(/Projects/i)).toBeInTheDocument();
    expect(screen.getByText(/Resume/i)).toBeInTheDocument();
  });

  it("should render the Desktop menu", () => {
    render(<Header />);
    const menu = screen.getByLabelText("desktopMenu");
    expect(menu).toBeInTheDocument();
  });

  it("should render Header", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
