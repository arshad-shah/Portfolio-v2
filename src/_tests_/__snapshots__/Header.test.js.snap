// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header of the site: should render Header 1`] = `
<div>
  <div
    class="Header-root css-1ev3ynq"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      id="back-to-top-anchor"
    />
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      data-testid="header"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 Header-title css-65a2cr-MuiTypography-root"
        >
          ARSHAD SHAH
        </h6>
        <div
          aria-label="desktopMenu"
          class="MuiBox-root css-yv9lcr"
          data-testid="desktopMenu"
        >
          <a
            aria-label="About Me"
            class="Desktop-menuItem"
            color="default"
            data-block="Aboutmemenuitemdesktop"
            data-testid="aboutmeOption"
          >
            <span
              class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
            >
              About Me
            </span>
          </a>
          <a
            aria-label="Projects"
            class="Desktop-menuItem"
            color="default"
            data-block="Projectsmenuitemdesktop"
            data-testid="projectsOption"
          >
            <span
              class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
            >
              Projects
            </span>
          </a>
          <a
            aria-label="Resume"
            class="Desktop-menuItem"
            color="default"
            data-block="Resumemenuitemdesktop"
            data-testid="resumeOption"
            href="ArshadResume.pdf"
          >
            <span
              class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
            >
              Resume
            </span>
          </a>
        </div>
      </div>
    </header>
  </div>
</div>
`;
